test/e2e/conformance/validatingadmissionpolicy_test.go

```diff
@@ -0,0 +1,179 @@
+					{
+						RuleWithOperations: admissionregistrationv1.RuleWithOperations{
+							Operations: []admissionregistrationv1.OperationType{
+								admissionregistrationv1.Create,
+								admissionregistrationv1.Update,
+							},
+							Rule: admissionregistrationv1.Rule{
+								APIGroups:   []string{wildwestv1alpha1.SchemeGroupVersion.Group},
+								APIVersions: []string{wildwestv1alpha1.SchemeGroupVersion.Version},
+								Resources:   []string{"cowboys"},
+							},
+						},
+					},
+				},
+			},
+			Validations: []admissionregistrationv1.Validation{{
+				Expression: "object.spec.intent != 'bad'",
+			}},
+		},
+	}
+	policy, err = kubeClusterClient.Cluster(ws1Path).AdmissionregistrationV1().ValidatingAdmissionPolicies().Create(ctx, policy, metav1.CreateOptions{})
+
+	require.NoError(t, err, "failed to create ValidatingAdmissionPolicy")
+
+	require.Eventually(t, func() bool {
+		p, err := kubeClusterClient.Cluster(ws1Path).AdmissionregistrationV1().ValidatingAdmissionPolicies().Get(ctx, policy.Name, metav1.GetOptions{})
+		if err != nil {
+			return false
+		}
+
+		// check if ValidatingAdmissionPolicy status has been updated
+		// and no type checking errors came up
+		return p.Generation == p.Status.ObservedGeneration && p.Status.TypeChecking != nil && len(p.Status.TypeChecking.ExpressionWarnings) == 0
+	}, wait.ForeverTestTimeout, 100*time.Millisecond)
+
+	t.Logf("Installing validating admission policy binding into the first workspace")
+	binding := &admissionregistrationv1.ValidatingAdmissionPolicyBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "binding-",
+		},
+		Spec: admissionregistrationv1.ValidatingAdmissionPolicyBindingSpec{
+			PolicyName:        policy.Name,
+			ValidationActions: []admissionregistrationv1.ValidationAction{admissionregistrationv1.Deny},
+		},
+	}
+
+	_, err = kubeClusterClient.Cluster(ws1Path).AdmissionregistrationV1().ValidatingAdmissionPolicyBindings().Create(ctx, binding, metav1.CreateOptions{})
+	require.NoError(t, err, "failed to create ValidatingAdmissionPolicyBinding")
+
+	badCowboy := wildwestv1alpha1.Cowboy{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "cowboy-",
+		},
+		Spec: wildwestv1alpha1.CowboySpec{
+			Intent: "bad",
+		},
+	}
+
+	goodCowboy := wildwestv1alpha1.Cowboy{
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "cowboy-",
+		},
+		Spec: wildwestv1alpha1.CowboySpec{
+			Intent: "good",
+		},
+	}
+
+	t.Logf("Verifying that creating bad cowboy resource in first logical cluster is rejected")
+	require.Eventually(t, func() bool {
+		_, err := cowbyClusterClient.Cluster(ws1Path).WildwestV1alpha1().Cowboys("default").Create(ctx, &badCowboy, metav1.CreateOptions{})
+		if err != nil {
+			if errors.IsInvalid(err) {
+				if strings.Contains(err.Error(), "failed expression: object.spec.intent != 'bad'") {
+					return true
+				}
+			}
+			t.Logf("Unexpected error when trying to create bad cowboy: %s", err)
```

Comment:
nit: a success is not unexpected while waiting for the policy to apply.

Reply:
This is within a check for `err != nil`, so it only logs any error that's not the policy rejection. If resource creation succeeds, it just returns `false` but doesn't log this line.